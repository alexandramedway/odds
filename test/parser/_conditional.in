do if true then 42 else 43
do if false then 42 else 43
do if 42 then "pass" else "fail"
do if 0 then "pass" else "fail"
do if 5 - 5 then true else false
do if true then 40 + 2 else 5 * 5
