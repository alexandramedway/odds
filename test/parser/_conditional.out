[
    Do(If(Bool_lit(true), Int_lit(42), Int_lit(43))) ;
    Do(If(Bool_lit(false), Int_lit(42), Int_lit(43))) ;
    Do(If(Int_lit(42), String_lit(pass), String_lit(fail))) ;
    Do(If(Int_lit(0), String_lit(pass), String_lit(fail))) ;
    Do(
        If(
            Binop(Int_lit(5), Sub, Int_lit(5)),
            Bool_lit(true),
            Bool_lit(false)
        )
    ) ;
    Do(
        If(
            Bool_lit(true),
            Binop(Int_lit(40), Add, Int_lit(2)),
            Binop(Int_lit(5), Mult, Int_lit(5))
        )
    )
]
